File: java/src/main/java/com/languageweaver/sdk/samples/translations/file/EdgeFileTranslationService.java
package com.languageweaver.sdk.samples.translations.file;

import com.languageweaver.sdk.common.constants.Format;
import com.languageweaver.sdk.common.constants.PdfConverter;
import com.languageweaver.sdk.common.edge.EdgeLanguageWeaverClient;
import com.languageweaver.sdk.translate.edge.request.EdgeTranslateFileRequest;
import com.languageweaver.sdk.translate.edge.result.EdgeTranslationFileResult;

import java.io.File;
import java.io.IOException;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;

public class EdgeFileTranslationService {

    public static void main(String[] args) throws Exception {
        try (EdgeLanguageWeaverClient lwClient = new EdgeLanguageWeaverClient().build()) {
            translateFile(lwClient);
            translateFileWithPdfConverter(lwClient);
            translateFileWithLinguisticOptions(lwClient);
        }
    }

    private static void translateFile(EdgeLanguageWeaverClient lwClient) throws IOException, InterruptedException {
        EdgeTranslateFileRequest translateFileRequest = new EdgeTranslateFileRequest()
                .setLanguagePairId("EngFra_Generic_SRV_TNMV_8_5_x_1")
                // provide full path to the source file
                .setInputFile(Paths.get("java", "src", "main", "resources", "input", "input1.txt").toFile().getAbsolutePath())
                .setOutputFile(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath() + File.separator + "input1-translated.txt")
                .setInputFormat(Format.PLAIN)
                .addDictionary("DictionaryName1")
                .addDictionary("DictionaryName2");

        final EdgeTranslationFileResult translateFile = lwClient.translateFile(translateFileRequest);
        // handle result
    }

    private static void translateFileWithPdfConverter(EdgeLanguageWeaverClient lwClient) throws IOException, InterruptedException {
        EdgeTranslateFileRequest translateFileRequest = new EdgeTranslateFileRequest()
                .setLanguagePairId("EngFra_Generic_SRV_TNMV_8_5_x_1")
                // provide full path to the source file
                .setInputFile(Paths.get("java", "src", "main", "resources", "input", "input1.pdf").toFile().getAbsolutePath())
                .setOutputFile(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath() + File.separator + "input1-translated.docx")
                .setInputFormat(Format.PDF)
                .setPdfConverter(PdfConverter.STANDARD)
                .addDictionary("DictionaryName1")
                .addDictionary("DictionaryName2");

        final EdgeTranslationFileResult translateFile = lwClient.translateFile(translateFileRequest);
        // handle result
    }

    private static void translateFileWithLinguisticOptions(EdgeLanguageWeaverClient lwClient) throws IOException, InterruptedException {
        Map<String, String> linguisticOptions = new HashMap<>();
        linguisticOptions.put("formality", "Informal");
        EdgeTranslateFileRequest translateFileRequest = new EdgeTranslateFileRequest()
                .setLanguagePairId("EngJpn_Generic_SRV_TNMV_8_5_x_6")
                // provide full path to the source file
                .setInputFile(Paths.get("java", "src", "main", "resources", "input", "input4.txt").toFile().getAbsolutePath())
                .setOutputFile(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath() + File.separator + "input4-translated.txt")
                .setLinguisticOptions(linguisticOptions)
                .setInputFormat(Format.PLAIN)
                .addDictionary("DictionaryName1")
                .addDictionary("DictionaryName2");

        final EdgeTranslationFileResult translateFile = lwClient.translateFile(translateFileRequest);
        // handle result
    }
}

File: dotnet/Samples/Translations/File/EdgeFileTranslationService.cs
using LanguageWeaver.Sdk.Common.Edge;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.Translate.Edge.Request;
using LanguageWeaver.Sdk.Translate.Edge.Result;

namespace LanguageWeaver.Sdk.Samples.Translations.File;

public class EdgeFileTranslationService
{
    public static void Main()
    {
        using var lwClient = new EdgeLanguageWeaverClient().Build();

        TranslateFile(lwClient);
        TranslateFileWithPdfConverter(lwClient);
        TranslateFileWithLinguisticOptions(lwClient);
    }

    private static void TranslateFile(EdgeLanguageWeaverClient lwClient)
    {
        var translateFileRequest = new EdgeTranslateFileRequest
        {
            LanguagePairId = "EngFra_Generic_SRV_TNMV_8_5_x_1",
            InputFormat = Format.Plain,
            Dictionaries = new List<string>
            {
                "DictionaryName1",
                "DictionaryName2"
            },
            // provide full path to the source and output file
            InputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input", "input1.txt"),
            OutputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output", "input1-translated.txt")
        };

        EdgeTranslationFileResult translateFileResult = lwClient.TranslateFile(translateFileRequest);
        // handle result
    }

    private static void TranslateFileWithPdfConverter(EdgeLanguageWeaverClient lwClient)
    {
        var translateFileRequest = new EdgeTranslateFileRequest
        {
            LanguagePairId = "EngFra_Generic_SRV_TNMV_8_5_x_1",
            InputFormat = Format.Pdf,
            PdfConverter = PdfConverter.STANDARD,
            Dictionaries = new List<string>
            {
                "DictionaryName1",
                "DictionaryName2"
            },
            // provide full path to the source and output file
            InputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input", "input1.pdf"),
            OutputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output", "input1-translated.docx")
        };

        EdgeTranslationFileResult translateFileResult = lwClient.TranslateFile(translateFileRequest);
        // handle result
    }

    private static void TranslateFileWithLinguisticOptions(EdgeLanguageWeaverClient lwClient)
    {
        Dictionary<string, string> linguisticOptions = new Dictionary<string, string>();
        linguisticOptions.Add("formality", "Formal");

        var translateFileRequest = new EdgeTranslateFileRequest()
        {
            LanguagePairId = "EngJpn_Generic_SRV_TNMV_8_5_x_6",
            LinguisticOptions = linguisticOptions,
            Dictionaries = new List<string>
            {
                "DictionaryName1",
                "DictionaryName2"
            },
            InputFormat = Format.Plain,
            // provide full path to the source and output file
            InputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input", "input1.txt"),
            OutputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output", "input1-translated.txt")
        };

        EdgeTranslationFileResult translateFileResult = lwClient.TranslateFile(translateFileRequest);
        // handle result
    }
}